app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,308 @@
+    const [error, setError] = useState<string | undefined>();
+
+    const [profileImage] = useState<undefined>();
+
+    const [updating, setUpdating] = useState(false);
+    const scrollViewRef = useRef<KeyboardAwareScrollView>();
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case 'update-profile':
+                        submitUser();
+                        break;
+                    case closeButtonId:
+                        close();
+                        break;
+                }
+            },
+        }, componentId);
+
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+
+        return () => {
+            unsubscribe.remove();
+            backHandler.remove();
+        };
+    }, [userInfo, isTablet, profileImage]);
+
+    useEffect(() => {
+        if (!isTablet) {
+            setButtons(componentId, {rightButtons: [rightButton]});
+        }
+    }, [isTablet]);
+
+    const style = getStyleSheet(theme);
+    const service = currentUser.authService;
+
+    const rightButton = {
+        id: 'update-profile',
+        enabled: false,
+        showAsAction: 'always' as const,
+        testID: 'edit_profile.save.button',
+        color: theme.sidebarHeaderTextColor,
+        text: intl.formatMessage({id: 'mobile.account.settings.save', defaultMessage: 'Save'}),
+    };
+
+    const close = () => {
+        if (isModal) {
+            dismissModal({componentId});
+        } else if (isTablet) {
+            DeviceEventEmitter.emit(Events.ACCOUNT_SELECT_TABLET_VIEW, '');
+        } else {
+            popTopScreen(componentId);
+        }
+
+        return true;
+    };
+
+    const enableSaveButton = (value: boolean) => {
+        if (isTablet) {
+            setCanSave(value);
+        } else {
+            const buttons = {
+                rightButtons: [{...rightButton, enabled: value}],
+            };
+
+            setButtons(componentId, buttons);
+        }
+    };
+
+    const submitUser = async () => {
+        enableSaveButton(false);
+        setError(undefined);
+        setUpdating(true);
+        try {
+            const partialUser: Partial<UserProfile> = {
+                email: userInfo.email,
+                first_name: userInfo.firstName,
+                last_name: userInfo.lastName,
+                nickname: userInfo.nickname,
+                position: userInfo.position,
+                username: userInfo.username,
+            };
+
+            const {error: reqError} = await updateMe(serverUrl, partialUser);
+
+            if (reqError) {
+                return resetScreen(reqError as Error);
+            }
+
+            close();
+        } catch (e) {
+            return resetScreen(e as Error);
+        }
+
+        return null;
```

Comment:
to avoid having to return null here you can do on L175 resetScreen(reqError as Error)

```suggestion
```